### PR TITLE
armv7m: fix DEBUG reset on exception logic

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-489

## Description

We want to reset on exception when not DEBUG

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by damianloew — DISMISSED** [review]

**Review by gerard5 — APPROVED** [review]
As for armv7m looks good to me. btw. I wonder if e.g. [armv7a](https://github.com/phoenix-rtos/plo/blob/52211471c2debc9878ff76057db0dc4b7a7ea64d/hal/armv7a/exceptions.c#L133-L137) doesn't require adding `#ifdef NDEBUG`.

**Comment by agkaminski** [comment]
@gerard5 yeah, let's add to all of them while at it

File: hal/ia32/exceptions.c [hunk 1]
```diff
@@ -158,6 +158,11 @@ static void exceptions_defaultHandler(unsigned int n, exc_context_t *ctx)
 	hal_consolePrint(buff);
 	hal_consolePrint("\033[0m");
 	hal_interruptsDisableAll();
+
+#ifdef NDEBUG
+	hal_cpuReboot();
+#endif
+
 	hal_cpuHalt();
```
Comment (gerard5) [inline hal/ia32/exceptions.c:166]: For safety `hal_cpuHalt();` may be wrapped in an endless loop like other targets.

**Review by gerard5 — APPROVED** [review]
Looks good to me. Verified on ia32, imxrt1176.

**Pull request merged**